### DISCUSSION TITLE
HASPone bug!

### DISCUSSION THREAD

**Discussion by rossbate** [HASwitchPlate/HASPone#20] [Q&A] [answered]
Found a bug.
Have temperature with icon on a button, then I have activate a page on that button. When changing to new page, page 9 the temperature icon is overlaid on the new page.

Also how can I get thermostat from Honeywell Evohome on that page? Can see a blueprint for thermostat??
![1615023855539990515750420986123](https://user-images.githubusercontent.com/39503487/110202440-8946a280-7e60-11eb-8166-3c7a911288f2.jpg)
![16150238697657030388776418053532](https://user-images.githubusercontent.com/39503487/110202446-96fc2800-7e60-11eb-8719-4bcf2879e94f.jpg)

**Comment by aderusha** [comment]
Are you using the Activate Page blueprint with that?  If os, I've just pushed a minor change to the behavior on that blueprint, can you try re-importing and see if it makes any difference?

**Reply by rossbate** [reply]
Yes using the blueprint for activate page. Will re-import and try it out. Thanks to all who have contributed to this project, only just discovered it and love it.

**Reply by eboston** [reply]
How do you re-import a blueprint?  Just use the IMPORT BLUEPRINT button again or is there some other way?

**Reply by bertmagin** [reply]
Yes,, just import it again

**Reply by eboston** [reply]
When I tried importing the core functionality blueprint, it reported the file already existed.

**Reply by rossbate** [reply]
Delete the old blueprint

**Reply by eboston** [reply]
okay.  What about existing automation based on the old blueprint?  Anything have to be done to them so they use the new blueprint?

**Reply by aderusha** [reply]
So long as the new blueprint retains the old blueprint's filename, you should be all set.

**Comment by aderusha** [comment, marked as answer]
I've just posted another update to "Activate Page" after doing some more testing.  Previously, the "activate page" was being triggered on button press (not release), so when you pressed the button down the page immediately changed.  The way that the overlay icons work is that they are triggered on release, so when you let up on the button, it paints the icon back into place (because pressing the button removes it).  So, press button, page changes, that makes you release the button, icon is painted on the new screen.

In general, changing pages on Nextion on button press is problematic, and can lead to buttons being "stuck" as they might never catch a release event, as you never actually let off the button because the page changed before you removed your finger.  So, step one was to fix that blueprint such that it triggers on release instead of press, which I should have done in the first place due to the issue above.

However... still doesn't fix the problem 100% of the time.  It becomes a race as to which of the two blueprints are going to trigger and then execute first, with both responding to the same button-release event.  After some testing, what I've "fixed" is to simply add a 100msec delay to the page change blueprint.  You release the button, home assistant catches that, repaints the icon right away, and then 0.1 seconds later sends the command to change the page.  In my testing this seems to work reliably while not interfering w/ icon overlays.

**Reply by IIAIronWolf** [reply]
Does this mean we will be getting an update and need to re-run the core functionality automation?

**Reply by IIAIronWolf** [reply]
I ask because I am not using a custom 'activate page' blueprint, only the core functionality as of right now.

**Reply by aderusha** [reply]
OK, so the OP was using the Activate Page blueprint.  Sounds like you are seeing some unexpected icon hanging out somewhere.  If you’re not using the Activate Page blueprint, can you explain a bit about what you’re seeing on your HASP?

**Reply by IIAIronWolf** [reply]
I am actually seeing the exact same issue as the OP on occasion, with the sticking icons, sometimes being the temperature icon. I am not using a activate page blueprint. I simply added the four blueprints you made in order to recreate the status page you had before HASPone, including date, time, temp, and weather condition. Therefore, either those blueprints have this same issue, or the Core Functionality blueprint may also need to be updated.
In addition, and as you mentioned in this first comment, the page buttons sometimes stick in the 'activated' color, but those were only setup by core functionality on my device. I have not customized page buttons. I have submitted issue #28

**Comment by IIAIronWolf** [comment]
Having the same issue here. This belongs in the Issues section though.

**Reply by aderusha** [reply]
Is the current version working for you?

**Reply by IIAIronWolf** [reply]
> Does this mean we will be getting an update and need to re-run the core functionality automation?

**Reply by aderusha** [reply]
You’ll want to re-import the Activate Page blueprint to address the issue in the OP.
